packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,342 @@
+
+Currently there is no ([legal](https://codesandbox.io/s/state-fill-usecase-42u1ev?file=/App.tsx)) way to achieve this because we only support Uncontrolled Dropdown where you can set default selection on first render, but every other selection is handled internally only.
+
+### Problem Statement 2: Reset SelectInput value
+
+If you have a "Reset" button in your form, there isn't way to de-select an already selected item ([legally](https://razorpay.slack.com/archives/C01H13RTF8V/p1681287510398639?thread_ts=1681286843.013449&cid=C01H13RTF8V)). For such usecases as well we need controlled Dropdown
+
+### Proposed Approaches
+
+<details>
+<summary><code>onChange</code> + <code>isSelected</code></summary>
+
+#### `onChange` + `isSelected`
+
+Slightly similar to primer and closest to HTML's `<select>`.
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput label="City" onChange={onSelectChange} />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem
+        // New Prop 👇🏼
+        isSelected={currentSelection === 'mumbai'}
+        title="Mumbai"
+        value="mumbai"
+      />
+      <ActionListItem
+        isSelected={currentSelection === 'bangalore'}
+        title="Bangalore"
+        value="bangalore"
+      />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+</details>
+
+<details>
+<summary><code>onChange</code> + <code>value</code></summary>
+
+#### `onChange` + `value`
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput
+    label="City"
+    // New Prop 👇🏼
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+</details>
+
+<details>
+<summary><code>useSelect</code> hook</summary>
+
+#### useSelect
+
+```jsx
+const { selectItem, reset, dropdownRef } = useSelect();
+
+const onSomeAction = () => {
+  selectItem('mumbai');
+};
+
+const onResetClick = () => {
+  reset();
+};
+
+<Dropdown ref={dropdownRef}>
+  <SelectInput />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+Inspirations
```

Comment:
Link? @saurabhdaware

Reply:
ops I missed this heading here. I moved the references to the end of doc